### PR TITLE
todo: issue 41161[Do not merge until v1.10]

### LINKED ISSUE #41161: Remove --require-kubeconfig and the default kubeconfig path in v1.10
This will be the end of the deprecation period for --require-kubeconfig.

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
todo: issue 41161

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #41161

**Special notes for your reviewer**:
Do not merge until v1.10

**Release note**:

```release-note
Remove --require-kubeconfig and the default kubeconfig path
```

### CODE REVIEW

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -117,8 +114,6 @@ type KubeletFlags struct {
 // NewKubeletFlags will create a new KubeletFlags with default values
 func NewKubeletFlags() *KubeletFlags {
 	return &KubeletFlags{
-		// TODO(#41161:v1.10.0): Remove the default kubeconfig path and --require-kubeconfig.
-		RequireKubeConfig:       false,
 		KubeConfig:              flag.NewStringFlag("/var/lib/kubelet/kubeconfig"),
```
Comment (mtaufen) [inline cmd/kubelet/app/options/options.go]: We should also remove the default kubeconfig path, per the comment.

File: cmd/kubelet/app/server.go [hunk 2]
```diff
@@ -270,9 +270,6 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 	// About to get clients and such, detect standaloneMode
 	standaloneMode := true
 	switch {
```
Comment (mtaufen) [inline cmd/kubelet/app/server.go]: now that there's only one case, might as well make this an if statement

File: cmd/kubelet/app/server.go [hunk 3]
```diff
@@ -364,8 +361,6 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) (err error) {
 
 		} else {
 			switch {
```
Comment (mtaufen) [inline cmd/kubelet/app/server.go]: can be an else if now

**Comment by mtaufen** [comment]
Why won't this PR reopen...?

**Comment by mtaufen** [comment]
Seems this is a GitHub thing: https://github.com/isaacs/github/issues/361
@zhangxiaoyu-zidif can you create a new PR please?

**Comment by zhangxiaoyu-zidif** [comment]
@mtaufen OK，no problem，I will send it soon. =) thanks

**Pull request closed without merging**